apps/studio/src/server/modules/user/user.router.ts

```diff
@@ -173,4 +175,22 @@ export const userRouter = router({
 
       return true
     }),
+
+  updateDetails: protectedProcedure
+    .input(updateDetailsInputSchema)
+    .output(updateDetailsOutputSchema)
+    .mutation(async ({ ctx, input: { name, phone } }) => {
+      // We don't have to check if the user is admin here
+      // because we only allow users to update their own details
+      // They should be able to update their own details even without any resource permissions
+
+      await db
+        .updateTable("User")
+        .where("id", "=", ctx.user.id)
+        .set({ name, phone })
+        .returningAll()
+        .executeTakeFirst()
+
+      return true
```

Comment:
[nit] should we minimally return the new properties to frontend?

Reply:
done!